### PR TITLE
Implement CSP for AdminWebClient AB#9196

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#9196](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9196)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description

Adds Content Security policy settings to Admin Web Client.
Updated documentation
Fixed  - minor smell in Communication.

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [X] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES | NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/AdminWebClient/README.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ Add the following text:
 {
   "OpenIdConnect": {
     "ClientSecret": "[Ask team for Client Secret]"
+  },
+  "ContentSecurityPolicy": {
+        "connect-src": "https://spt.apps.gov.bc.ca/com.snowplowanalytics.snowplow/tp2 https://dev.oidc.gov.bc.ca/ http://localhost:*/  http:/[YOUR IP]:*/ ws://[YOUR IP]:*/",
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/README.md:20]: I dont think we have snowplow in admin

File: Apps/AdminWebClient/src/appsettings.json [hunk 2]
```diff
@@ -44,5 +44,12 @@
         "Enabled": "true",
         "BasePath": "/admin",
         "IPs": []
+    },
+    "ContentSecurityPolicy": {
+        "connect-src": "https://spt.apps.gov.bc.ca/com.snowplowanalytics.snowplow/tp2 https://oidc.gov.bc.ca/",
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/appsettings.json:49]: snowplow?

File: Apps/AdminWebClient/src/appsettings.json [hunk 3]
```diff
@@ -44,5 +44,12 @@
...
+        "frame-src": "https://oidc.gov.bc.ca/",
+        "script-src": "https://www2.gov.bc.ca/StaticWebResources/static/sp/sp-2-14-0.js 'sha256-q+lOQm0t+vqQq4IdjwI4OwRI9fKfomdxduL1IJYjkA4='",
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/appsettings.json:51]: www2.gov.bc.ca/StaticWebResources/ is for snowplow

File: Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs [hunk 4]
```diff
@@ -439,9 +439,11 @@ public void UseContentSecurityPolicy(IApplicationBuilder app)
             string connectSrc = cspSection.GetValue<string>("connect-src", string.Empty);
             string frameSrc = cspSection.GetValue<string>("frame-src", string.Empty);
             string scriptSrc = cspSection.GetValue<string>("script-src", string.Empty);
+            string styleSrc = cspSection.GetValue<string>("style-src", string.Empty);
+            string fontSrc = cspSection.GetValue<string>("font-src", string.Empty);
```
Comment (FuriousLlama) [inline Apps/Common/src/AspNetConfiguration/StartupConfiguration.cs:443]: this is kinda odd. If the weblcient does not need it, then admin probably doesn't need it either

**Review by tiagograf-AOT — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**